pkg/test/framework/testcontext.go

```diff
@@ -60,11 +60,7 @@ type TestContext interface {
 	// Methods for interacting with the underlying *testing.T.
 	Error(args ...interface{})
 	Errorf(format string, args ...interface{})
```

Comment:
FYIL With this, we won't be able to run the same test multiple times with the same T.

Reply:
I'm not sure I understand the use case ... can you clarify?

Reply:
If we ever want to rerun flaky tests, we will need to hide T inside TestContext and only fail when all of the reruns fail. Exposing T disallows that. If we rerun the case in case of a flake, the test will still show up as failed in report.

Which is probably not a high priority thing, but I thought I'd call it out before lgtm'ing.

Reply:
@ozevren It's still hidden ... it's exposed only by `testContext`, not the interface `TestContext`.

Reply:
If we ever want to do the flake rerun, however, we will need to revert what you're doing here, right? We can't interject/capture the call.

/lgtm
I am lgtm'ing to unblock, but something to think about.

Reply:
Ah ok, I get what you're saying now.  This change doesn't prevent us from optionally implementing `testing.T` methods one-by-one to intercept before delegating to `T` ... in fact, I'm already doing that with `Parallel`.  Of course as soon implement a method,  we're back to the original call stack problem.  This PR fixes the issue for now, but we may break it again in the future, depending on how retries are implemented.

Reply:
@ozevren PTAL I think I managed to figure out a way to make this work.  Overriding all of the methods **except** `Helper`.

Reply:
/lgtm